peers/external_handler.go

```diff
@@ -41,6 +42,7 @@ type expectedResponses struct {
 // Teleporter application relayer, as well as handle timeouts.
 func NewRelayerExternalHandler(
 	logger logging.Logger,
+	metrics *AppRequestNetworkMetrics,
 	registerer prometheus.Registerer,
```

Comment:
It's odd that we pass in an actual metrics collector alongside an independent registerer (that we have hooked up to the noop registerer). Instead, how about we remove `registerer` as a parameter altogether, and hardcode `prometheus.DefaultRegisterer` in the call to `NewAdaptiveTimeoutManager`. We could also pass in the timeout manager as an arg.

Reply:
I agree, I thought it was odd but not that big of a deal. I went with the first option above in [e047507](https://github.com/ava-labs/awm-relayer/pull/461/commits/e047507aef1396b48719d1e047436dc082105482)